main_test.go

```diff
@@ -0,0 +1,316 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/testcontainers/testcontainers-go"
+)
+
+const exporterEndpointKey = "OTEL_EXPORTER_OTLP_ENDPOINT"
+
+var originalEnvVar string
+var originalServiceNameFlag string
+var originalEndpoint string
+
+func init() {
+	originalEndpoint = os.Getenv(exporterEndpointKey)
+	originalEnvVar = os.Getenv("OTEL_SERVICE_NAME")
+	originalServiceNameFlag = serviceNameFlag
+}
+
+type TestReader struct{}
+
+func (tr *TestReader) Read() ([]byte, error) {
+	b, err := ioutil.ReadFile("TEST-sample.xml")
+	if err != nil {
+		return []byte{}, err
+	}
+
+	return b, nil
+}
+
+type TestTraceAttributeValue struct {
+	IntValue    string `json:"intValue"`
+	StringValue string `json:"stringValue"`
+}
+
+type TestTraceAttribute struct {
+	Key   string                  `json:"key"`
+	Value TestTraceAttributeValue `json:"value,omitempty"`
+}
+
+// TestTraceReport holds references to the File exporter defined by the otel-collector
+type TestTraceReport struct {
+	ResourceSpans []struct {
+		Resource struct {
+			Attributes []TestTraceAttribute `json:"attributes"`
+		} `json:"resource"`
+		InstrumentationLibrarySpans []struct {
+			InstrumentationLibrary struct {
+				Name string `json:"name"`
+			} `json:"instrumentationLibrary"`
+			Spans []struct {
+				TraceID           string               `json:"traceId"`
+				SpanID            string               `json:"spanId"`
+				ParentSpanID      string               `json:"parentSpanId"`
+				Name              string               `json:"name"`
+				Kind              string               `json:"kind"`
+				StartTimeUnixNano string               `json:"startTimeUnixNano"`
+				EndTimeUnixNano   string               `json:"endTimeUnixNano"`
+				Attributes        []TestTraceAttribute `json:"attributes"`
+				Status            struct {
+				} `json:"status"`
+			} `json:"spans"`
+		} `json:"instrumentationLibrarySpans"`
+	} `json:"resourceSpans"`
+}
```

Comment:
Struct objects mapped from the exporter. I understand that this struct could change if the otel-exporter is changed